docs/design/multi_slo_doc_support.md

```diff
@@ -0,0 +1,102 @@
+
+This is consistant with most services today, but some services do in fact have multiple SLO documents
+
+We want to properly enable the usage pattern of defining more-than-one SLO documents for any given service. We want to support 'service' and 'slo-document' having a one-to-N relationship (where N can be zero or any positive integer).
+
+### Examples
+
+The service 'cincinnati' [has **a single** SLO-document](https://gitlab.cee.redhat.com/service/app-interface/-/tree/1b368998a913be3f2d45a13934eb6403d4e751de/data/services/cincinnati/slo-documents). It has [three SLOs](https://gitlab.cee.redhat.com/service/app-interface/-/blob/1b368998a913be3f2d45a13934eb6403d4e751de/data/services/cincinnati/slo-documents/cincinnati.yml#L13-47) defined. These can be seen correctly formatted in its [report YML doc](https://gitlab.cee.redhat.com/service/app-interface/-/blob/fae9d83da7db25c2ec5364eb4a6ebe8ee542e845/data/reports/Cincinnati/2021-08-01.yml#L63-78) and its [visual report page](https://visual-app-interface.devshift.net/reports#/reports/Cincinnati/2021-08-01.yml). **This is working as intended.**
+
+The service 'ocm' [has **two** SLO-documents](https://gitlab.cee.redhat.com/service/app-interface/-/tree/1b368998a913be3f2d45a13934eb6403d4e751de/data/services/ocm/slo-documents). Each of these documents define SLOs with the same set of names (latency, errors, availability). There are a total of 6 SLOs defined for this serice (2 docs * 3 SLOs-per-doc = 6; 2 SLOs named 'latency', 2 SLOs named 'errors', and 2 SLOs named 'availability'). However, the 'ocm' [visual report page](https://visual-app-interface.devshift.net/reports#/reports/ocm/2021-08-01.yml) only renders 3 SLOs. **This is NOT working as intended.** The tools do not take into account the possibility of a service defining the same SLO name multiple times across multiple docs for the same service. Presumably the SLO datapoints in dash.db are incorrectly overwriting one another or are incorrectly being aggregated together.
+
+The service ['service-registry'](https://gitlab.cee.redhat.com/service/app-interface/-/tree/6762ce9ae3c34abedc21abf768437a5e7753652a/data/services/service-registry/slo-documents) would have [a similar problem](https://visual-app-interface.devshift.net/reports#/reports/service-registry/2021-08-01.yml)(although its visual report page fails to render for a seemingly-unrelated-reason) to 'ocm'.
+
+## Proposal
+
+### Overview
+
+The 'name' ([example](https://gitlab.cee.redhat.com/service/app-interface/-/blob/8bba50902109207d7e8a0b8f856bec92ede1e482/data/services/ocm/slo-documents/accounts-mgmt.yml#L7)) of these SLO docs will be introduced as an identifier for the SLO metric data stored in dashdotdb.
+
+SLO data will be stored in report-YML files in the following format:
+```
+service_slo:
+  slo_doc_name1:
```

Comment:
Is there a desire to split out the name of the slo like this instead of just having service_slo be a list similar to what it is in app-interace?

Reply:
@rrati 

Can you give an example of the output you're proposing?

Like this?

```
service_slo:
  - cluster: app-sre-prod-01
    slo_doc_name: some-doc-name-1
    namespace: app-interface-production
    slo_name: Reconciliation time of app-interface integrations
    slo_value: 27.0
    slo_target: 95.0
  - cluster: app-sre-prod-01
    slo_doc_name: some-doc-name-2
    namespace: app-interface-production
    slo_name: Reconciliation time of app-interface integrations
    slo_value: 27.0
    slo_target: 95.0
  - ...
```

If so that should be fine.

Reply:
Yep, something like that is what I'm proposing.

Reply:
Cool. I think I prefer that too.